### ISSUE TITLE
Document new features in ASP.NET Core 1.1

### ISSUE THREAD

**Issue by tdykstra** [aspnet/Docs#2260]
Features highlighted in the [announcement blog post](https://blogs.msdn.microsoft.com/webdev/2016/11/16/announcing-asp-net-core-1-1/):

- [x] Configuration Rick-Anderson
- [x] URL Rewriting Middleware @GuardRex
- [x] Response Caching Middleware @GuardRex 
- [x] Response Compression Middleware #2279  @GuardRex 
- [x] WebListener tdykstra
- [x] View Components as Tag Helpers Rick-Anderson
- [x] Middleware as MVC filters tdykstra
- [x] Cookie-based TempData provider Rick-Anderson
- [x] View compilation
- [x] Azure App Service logging provider tdykstra
- [x] Azure Key Vault configuration provider 
- [x] Redis and Azure Storage Data Protection Key Repositories

**Comment by GuardRex** [comment]
Missing Response Compression Middleware?

**Comment by Rick-Anderson** [comment]
@GuardRex  2nd item, i've updated it.

**Comment by GuardRex** [comment]
https://blogs.msdn.microsoft.com/webdev/2016/10/25/announcing-asp-net-core-1-1-preview-1/

The compression (gzipping) bits are a separate middleware from the caching middleware.

**Comment by GuardRex** [comment]
... according to my crappy testing here, it's about 30% slower than IIS, but they said its a WIP and that they'll get it faster later.

https://github.com/aspnet/BasicMiddleware/tree/dev/src/Microsoft.AspNetCore.ResponseCompression

https://github.com/aspnet/BasicMiddleware/issues/179

**Comment by GuardRex** [comment]
... and I guess since we're on the subject. If you decide to add either or both "Response Caching" and/or "Response Caching Middleware" (it seems like this would be one doc tho) to Performance docs and looking for a volunteer, let me know.

**Comment by Rick-Anderson** [comment]
[Response Caching](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response) is done, but it would be great if you could do [Response caching middleware](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/middleware)

**Comment by GuardRex** [comment]
I mistyped it initially. I was thinking "compression" and typed "caching" by mistake. Yes, thanks, I'll float you an outline this afternoon on the caching middleware doc.

If you decide to do a Response Compression one, as well, I like that topic, so let me know if you would like an outline on that one, too.

**Comment by GuardRex** [comment]
Sorry ... didn't see the end of your msg. I'll wait on the outline.

**Comment by GuardRex** [comment]
@Rick-Anderson You have me listed for "Response Compression" but I'm working on "URL Rewriting" per https://github.com/aspnet/Docs/issues/2304.

**Comment by GuardRex** [comment]
@tdykstra 

[@]Rick-Anderson let me have the Response Compression doc https://github.com/aspnet/Docs/issues/2279#issuecomment-270262068.

The Response Caching Middleware doc is also on me (it's awaiting merge https://github.com/aspnet/Docs/pull/2315).

**Comment by GuardRex** [comment]
@tdykstra @Rick-Anderson

![oliver](https://cloud.githubusercontent.com/assets/1622880/21904341/e0d7a1a8-d8c8-11e6-8be9-353b2146eb6e.png)
*"Please sir, I want some more."* - Oliver Twist (Mark Lester), *Oliver!*, 1968

I just put the PR in for *Response Compression Middleware*. I'm lightly working on [Using IIS Modules with ASP.NET Core](https://github.com/aspnet/Docs/issues/2367), but that's not going to take much time. I'm waiting for team feedback and will put the doc together quickly thereafter.

Do you have a needed "up for grabs" lying around?

**Comment by Rick-Anderson** [comment]
Are you interested #1993 ? It has a really good outline. I think you'd just use the outline without too much change.

**Comment by GuardRex** [comment]
Excellent! ... I'm on the case, and working with @anurse is a bonus!

**Comment by Rick-Anderson** [comment]
you mean [@]anurse :)

**Comment by GuardRex** [comment]
Well ...... I admit that I did want him to know I'm excited you suggested this one to me. It's an honor, really.

**Comment by anurse** [comment]
![http://weknowmemes.com/generator/uploads/generated/g1441151759445560028.jpg](http://weknowmemes.com/generator/uploads/generated/g1441151759445560028.jpg)

**Comment by GuardRex** [comment]
URL Rewriting Middleware - :+1: done https://github.com/aspnet/Docs/pull/2446
Response Compression Middleware - :+1: done https://github.com/aspnet/Docs/pull/2505
View Components as Tag Helpers - ? - https://github.com/aspnet/Docs/issues/1757
View Compilation - :+1: done https://github.com/aspnet/Docs/pull/2499
Azure Key Vault configuration provider - **almost done** :smile: https://github.com/aspnet/Docs/pull/2531
Redis and Azure Storage Data Protection Key Repositories - :+1: done https://github.com/aspnet/Docs/pull/2510

**Comment by GuardRex** [comment]
I guess this is as good a place as any to ask about these ...

### Azure Integration
[Azure Integration](https://github.com/aspnet/AzureIntegration) seems lightly covered. It's only covered in a Note in the Fundamentals/Logging doc. It reminds me of a notion that I've had occasionally ...

When it comes to the basics for ASP.NET Core on Azure Apps, the two existing docs, [Deploy an ASP.NET Core web app to Azure using Visual Studio](https://docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-azure-webapp-using-vs) and [Publishing to an Azure Web App with continuous deployment](https://docs.microsoft.com/en-us/aspnet/core/publishing/azure-continuous-deployment), seem like overkill (e.g., CI, Git, VS, EF, SQL Database).

Are you sure you don't want a *Publishing to Azure App Service* doc? ... akin to *Publishing to IIS*. Such a doc would be an ideal place to include Azure Integration. The [@]pakrym [sample](https://github.com/aspnet/AzureIntegration/tree/master/sample/AzureAppServicesSample) would be nice in that context.

### Publishing to a Windows Virtual Machine on Azure
The *Publishing to IIS* doc covers most of that scenario. Other than spinning up the VM and a couple of minor deployment details, there isn't much else to say.

**Comment by Rick-Anderson** [comment]
@GuardRex  can you copy/paste that into a new issue to suggest that topic. Sounds like a good idea to me.
